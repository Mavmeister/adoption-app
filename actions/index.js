import URLs from '../constants/URLs'

export const ANIMALS_REQUEST = 'ANIMALS_REQUEST';
export const ANIMALS_OK = 'ANIMALS_OK';
export const ANIMALS_SAVE = 'ANIMALS_SAVE';
export const SETTINGS_REQUEST = 'SETTINGS_REQUEST';
export const SETTINGS_UPDATE = 'SETTINGS_UPDATE';
export const SETTINGS_OK = 'SETTINGS_OK';

const animalsRequest = value => ({
  type: ANIMALS_REQUEST,
});
const animalsOk = value => ({
  type: ANIMALS_OK,
  payload: value,
});
const animalsSave = value => ({
  type: ANIMALS_SAVE,
  payload: value,
});
const settingsRequest = value => ({
  type: SETTINGS_REQUEST,
  payload: value,
});
const settingsOk = value => ({
  type: SETTINGS_OK,
  payload: value,
});
const settingsUpdate = value => ({
  type: SETTINGS_UPDATE,
  payload: value,
});

const fetchAnimals = (args = {}) => (dispatch) => {
    dispatch(animalsRequest());
    return fetch(URLs.animals)
      .then(res => res.json())
      .then(json => {
        dispatch(animalsOk(json))
      })
}
const saveAnimals = (args = {}) => (dispatch) => {
  dispatch(animalsSave(args));
}

const fetchSettings = (args = {}) => (dispatch) => {
    dispatch(settingsRequest());
    return fetch(URLs.settings)
      .then(res => res.json())
      .then(json => {
        dispatch(settingsOk(json))
      })
}

const updateSettings = (args = {}) => (dispatch, getState) => {
  const state = getState();
  const newSettings = {...state.data.settings, ...args}
  dispatch(settingsUpdate(newSettings));
}


export {
  fetchAnimals,
  saveAnimals,
  fetchSettings,
  updateSettings
}