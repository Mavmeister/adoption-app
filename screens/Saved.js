import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  View,

} from 'react-native';

// Sample data for dev
import data from '../data.json';
import SavedList from '../components/SavedList';
import { WebBrowser } from 'expo';

export default class SavedScreen extends React.Component {
  render() {
    return (
      <View style={styles.content}>
        <SavedList data={data} onPressItem={(id) => console.log(id)}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  content: {
    flex: 1,
  },
});
