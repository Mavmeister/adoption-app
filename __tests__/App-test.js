import 'react-native';
import * as actions from '../actions';
import rootReducer, { initialState } from '../reducers';
import React from 'react';
import Saved from '../screens/Saved';
import { store } from '../store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';


test('Saved renders correctly', () => {
  const tree = renderer.create(
  <Provider store={store}>
    <Saved />
  </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Returns initial state', () => {
  expect(rootReducer(initialState, {type: "TEST"})).toEqual({
    "data": {
      "animals": [],
      "loading": false,
        "savedAnimals": [],
        "settings": {
          "ageRange": {
            "max": 10,
            "min": 9,
          },
          "id": 1001,
          "profile": "",
          "typePreference": "cat",
        },
    }
  })
});

// describe('Actions', () => {
//   it('Should fetch animals', () => {
//     const expectedAction = {
//       type: actions.ANIMALS_REQUEST
//     };
//     expect(actions.fetchAnimals()).toEqual(expectedAction)
//   })
// })
// describe('Reducers', () => {
//   it('Should return the initial state', () => {
//     expect(rootReducer(undefined, {}).toEqual({}))
//   })
// })