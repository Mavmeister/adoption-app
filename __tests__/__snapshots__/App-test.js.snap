// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saved renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffcd99",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#abe0e8",
          "flex": 1,
          "height": 1334,
          "padding": 15,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 667,
              "marginBottom": 10,
              "marginTop": 10,
              "width": 740,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "black",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
            "paddingBottom": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "lineHeight": 30,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 26,
              "lineHeight": 30,
            }
          }
        >
           years
        </Text>
        <Text
          style={
            Object {
              "fontSize": 22,
              "lineHeight": 30,
            }
          }
        >
          Female
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 30,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 20,
            "width": 200,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "black",
                  },
                ]
              }
            >
              Close Profile
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
